src/amplitude.js

```diff
@@ -680,104 +842,110 @@ Amplitude.prototype.logRevenue = function(price, quantity, product) {
 };
 
 /**
- * Remove events in storage with event ids up to and including maxEventId. Does
- * a true filter in case events get out of order or old events are removed.
+ * Remove events in storage with event ids up to and including maxEventId.
+ * @private
  */
-Amplitude.prototype.removeEvents = function (maxEventId, maxIdentifyId) {
-  if (maxEventId >= 0) {
-    var filteredEvents = [];
-    for (var i = 0; i < this._unsentEvents.length; i++) {
-      if (this._unsentEvents[i].event_id > maxEventId) {
-        filteredEvents.push(this._unsentEvents[i]);
-      }
-    }
-    this._unsentEvents = filteredEvents;
+Amplitude.prototype.removeEvents = function removeEvents(maxEventId, maxIdentifyId) {
+  _removeEvents(this, '_unsentEvents', maxEventId);
+  _removeEvents(this, '_unsentIdentifys', maxIdentifyId);
```

Comment:
Not sure why "this" needs to be passed in separately here as a scope. Can we just pass in ```this._unsentEvents``` and ```this._unsentIdentifys```?